### PR TITLE
Misc. TF-M+54L support

### PR DESCRIPTION
**Pull request by SebastianBoe**

### CODE REVIEW

File: tests/tfm/tfm_psa_test/CMakeLists.txt [hunk 1]
```diff
@@ -100,7 +100,7 @@ ExternalProject_Add(tfm_psa_arch_test_app
                 -DCROSS_COMPILE=${TFM_TOOLCHAIN_PATH}/${TFM_TOOLCHAIN_PREFIX}
                 -DPSA_TOOLCHAIN_FILE=${TFM_BINARY_DIR}/api_ns/cmake/${TFM_TOOLCHAIN_NS_FILE}
                 -DQCBOR_PATH${QCBOR_PATH_TYPE}=${CONFIG_TFM_QCBOR_PATH}
-                -DCMAKE_BUILD_TYPE=RelWithDebInfo
+                -DCMAKE_BUILD_TYPE=Debug
```
Comment (mswarowsky) [inline tests/tfm/tfm_psa_test/CMakeLists.txt:103]: I'm no fully sure about the consequences but is it smart to build always in Debug mode ? or only if specifically enabled?

**Review by mswarowsky — COMMENTED** [review]
(only reviewed commits that are not part of #14463)

There are two manifest commits updating tf-m the first one could be removed....

File: tests/tfm/tfm_psa_test/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply tests/tfm/tfm_psa_test/CMakeLists.txt:103]: hmm, I suppose it is unfortunate to test the debug variant and not the production variant.

Will see what the difference is.

File: tests/tfm/tfm_psa_test/testcase.yaml [hunk 2]
```diff
@@ -1,54 +1,75 @@
...
-  tfm.psa_test_internal_trusted_storage_lvl2:
-    tags: tfm_lvl2
-    extra_args: "CONFIG_TFM_PSA_TEST_INTERNAL_TRUSTED_STORAGE=y"
   tfm.psa_test_storage_lvl1:
     tags: tfm_lvl1
     extra_args: "CONFIG_TFM_PSA_TEST_STORAGE=y CONFIG_TFM_ISOLATION_LEVEL=1"
     timeout: 130
+    platform_allow: >
+      nrf5340dk_nrf5340_cpuapp_ns
+      nrf9160dk_nrf9160_ns
```
Comment (stephen-nordic) [inline tests/tfm/tfm_psa_test/testcase.yaml:17]: Need `nrf54l15pdk_nrf54l15_cpuapp_ns` added to `platform_allow` as well

**Comment by SebastianBoe** [comment]
reorganizing

**Pull request closed without merging**